Add multi-stage scenarios in Access Review API beta (#15387)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/accessreviewstagesettings.md

---
title: "accessReviewStageSettings resource type"
description: "In Azure AD access reviews, the accessReviewStageSettings represents the settings of the stages associated with a multi-stage access review."
author: "isabelleatmsft"
ms.localizationpriority: medium
ms.prod: "governance"
doc_type: resourcePageType
---

# accessReviewStageSettings resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

[!INCLUDE [accessreviews-disclaimer-v2](../../includes/accessreviews-disclaimer-v2.md)]

Represents the settings of the stages associated with a [multi-stage access review](accessreviewscheduledefinition.md) object. 

## Properties
|Property|Type|Description|
|:---|:---|:---|

> |stageId|String|Unique identifier of the **accessReviewStageSettings**. The stageId will be used in **dependsOn** property to indicate the stage relationship. Required. |

Reviewer: Faith:
Please reorder this table by property name in alphabetical order

Missing recommendationInsightSettings property

After reviewing the new example 4 in POST accessReviewScheduleDefinition, I think we need a little more clarity on stageId and dependsOn. From the example, it seems stageId is a number you assign to the stage, and dependsOn indicates the preceding stageId. Trouble is, stageId is a String, leaving room for too much choice.

Can alphabets work? What about decimals?
If only integer numbers are allowed/recommended, please indicate in the description that stageId is a numeric value between 1-3 (or is it 0-2?, 0-3?).
For dependOn, please append examples to illustrate dependency on one stage and sequential order; "For example, "1" or "1", "2"" (please correct this if it's wrong)

> |dependsOn|String collection|Define the sequential or parallel order of each stageSettings. Currently only support single dependency|

Reviewer: ```suggestion
|dependsOn|String collection|Define the sequential or parallel order of each stageSettings. Only sequential stages are currently supported.|
```

|decisionsThatWillMoveToNextStage|String collection|Indicate which decisions will go to the next stage. Can be a sub-set of "Approve", "Deny", "Recommendation", or "NotReviewed". If not provided, all decisions will go to the next stage.|
|recommendationsEnabled|Boolean|Indicates whether showing recommendations to reviewers is enabled.|
| recommendationLookBackDuration | Duration| Optional field. Indicates the time period of inactivity (with respect to the start date of the review instance) that recommendations will be configured from. The recommendation will be to `deny` if the user is inactive during the look back duration. For reviews of groups and Azure AD roles, any duration is accepted. For reviews of applications, 30 days is the maximum duration. If not specified, the duration is 30 days. |
|reviewers|[accessReviewReviewerScope](../resources/accessreviewreviewerscope.md) collection|Defines who the reviewers are. If none are specified, the review is a self-review (users review their own access).  For examples of options for assigning reviewers, see [Assign reviewers to your access review definition using the Microsoft Graph API](/graph/accessreviews-reviewers-concept). |

> |fallbackReviewers|[accessReviewReviewerScope](../resources/accessreviewreviewerscope.md) collection|If provided, the fallback reviewers are asked to complete a review if the primary reviewers do not exist. For example, if managers are selected as `reviewers` and a principal under review does not have a manager in Azure AD, the fallback reviewers are asked to review that principal.|

Reviewer: Faith:
Please reorder this table by property name in alphabetical order

Missing recommendationInsightSettings property

After reviewing the new example 4 in POST accessReviewScheduleDefinition, I think we need a little more clarity on stageId and dependsOn. From the example, it seems stageId is a number you assign to the stage, and dependsOn indicates the preceding stageId. Trouble is, stageId is a String, leaving room for too much choice.

Can alphabets work? What about decimals?
If only integer numbers are allowed/recommended, please indicate in the description that stageId is a numeric value between 1-3 (or is it 0-2?, 0-3?).
For dependOn, please append examples to illustrate dependency on one stage and sequential order; "For example, "1" or "1", "2"" (please correct this if it's wrong)

---
title: "accessReviewStageSettings resource type"
description: "In Azure AD access reviews, the accessReviewStageSettings represents the settings of the stages associated with a multi-stage access review."
author: "isabelleatmsft"
ms.localizationpriority: medium
ms.prod: "governance"
doc_type: resourcePageType
---

# accessReviewStageSettings resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

[!INCLUDE [accessreviews-disclaimer-v2](../../includes/accessreviews-disclaimer-v2.md)]

Represents the settings of the stages associated with a [multi-stage access review](accessreviewscheduledefinition.md) object. 

## Properties
|Property|Type|Description|
|:---|:---|:---|
|decisionsThatWillMoveToNextStage|String collection|Indicate which decisions will go to the next stage. Can be a sub-set of "Approve", "Deny", "Recommendation", or "NotReviewed". If not provided, all decisions will go to the next stage. Optional. |

> |dependsOn|String collection|Define the sequential or parallel order of each stageSettings. Only sequential stages are currently supported. Required. |

Reviewer: ```suggestion
|dependsOn|String collection|Define the sequential or parallel order of each stageSettings. Only sequential stages are currently supported. For example, if stageId is "2", then dependsOn must be "1". If stageId is "1", do not specify dependsOn. Required. |
```

|recommendationsEnabled|Boolean|Indicates whether showing recommendations to reviewers is enabled. Required.|
| recommendationInsightsSettings|||
| recommendationLookBackDuration | Duration| Optional field. Indicates the time period of inactivity (with respect to the start date of the review instance) that recommendations will be configured from. The recommendation will be to `deny` if the user is inactive during the look back duration. For reviews of groups and Azure AD roles, any duration is accepted. For reviews of applications, 30 days is the maximum duration. If not specified, the duration is 30 days. |
|reviewers|[accessReviewReviewerScope](../resources/accessreviewreviewerscope.md) collection|Defines who the reviewers are. If none are specified, the review is a self-review (users review their own access).  For examples of options for assigning reviewers, see [Assign reviewers to your access review definition using the Microsoft Graph API](/graph/accessreviews-reviewers-concept). |

> |stageId|String|Unique identifier of the **accessReviewStageSettings**. The stageId will be used in **dependsOn** property to indicate the stage relationship. Required. |

Reviewer: ```suggestion
|stageId|String|Unique identifier of the **accessReviewStageSettings**. The stageId will be used in **dependsOn** property to indicate the stage relationship. We recommend using numeric values 1, 2, and 3. Required. |
```
